Intermittent integration test failure: targeting tests

Just saw this locally, I'm guessing it's trying to load the list of all targeting configs before the server is ready.

```
====================================================================================== ERRORS =======================================================================================
__________________________________________________________________ ERROR collecting nimbus/test_remote_settings.py __________________________________________________________________
nimbus/test_remote_settings.py:31: in <module>
    ???
nimbus/test_remote_settings.py:13: in load_data
    data = requests.post(
.tox/integration-test-nimbus/lib/python3.8/site-packages/requests/models.py:910: in json
    return complexjson.loads(self.text, **kwargs)
/usr/lib/python3.8/json/__init__.py:357: in loads
    return _default_decoder.decode(s)
/usr/lib/python3.8/json/decoder.py:337: in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
/usr/lib/python3.8/json/decoder.py:355: in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
E   json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
---------------------------------------------------------------------------------- Captured stderr ----------------------------------------------------------------------------------
/code/app/tests/integration/.tox/integration-test-nimbus/lib/python3.8/site-packages/urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host 'nginx'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
_______________________________________________________________________________ ERROR collecting gw1 ________________________________________________________________________________
Different tests were collected between gw0 and gw1. The difference is:
--- gw0

+++ gw1

@@ -1,14 +0,0 @@

-nimbus/test_remote_settings.py::test_check_targeting[DESKTOP-]
-nimbus/test_remote_settings.py::test_check_targeting[DESKTOP-first_run]
-nimbus/test_remote_settings.py::test_check_targeting[DESKTOP-first_run_chrome]
-nimbus/test_remote_settings.py::test_check_targeting[DESKTOP-first_run_win1903]
-nimbus/test_remote_settings.py::test_check_targeting[DESKTOP-urlbar_firefox_suggest]
-nimbus/test_remote_settings.py::test_check_targeting[DESKTOP-mac_only]
-nimbus/test_remote_settings.py::test_check_targeting[DESKTOP-no_enterprise_users]
-nimbus/test_remote_settings.py::test_check_targeting[DESKTOP-infrequent_user_uris]
-nimbus/test_remote_settings.py::test_check_targeting[DESKTOP-infrequent_win_user_uris]
-nimbus/test_remote_settings.py::test_check_targeting[DESKTOP-casual_user_uris]
-nimbus/test_remote_settings.py::test_check_targeting[DESKTOP-infrequent__or_casual_win_user_uris]
-nimbus/test_remote_settings.py::test_check_targeting[DESKTOP-regular_user_uris]
-nimbus/test_remote_settings.py::test_check_targeting[DESKTOP-core_user_uris]
-nimbus/test_remote_settings.py::test_check_targeting[DESKTOP-no_enterprise_or_past_vpn]
============================================================================== short test summary info ==============================================================================
ERROR nimbus/test_remote_settings.py - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
ERROR gw1
================================================================================= 2 errors in 1.63s =================================================================================
ERROR: InvocationError for command /code/app/tests/integration/.tox/integration-test-nimbus/bin/pytest --verify-base-url --base-url https://nginx/nimbus/ --self-contained-html --driver Firefox nimbus/ -k DESKTOP -m run_parallel -n 2 -vvv (exited with code 1)
______________________________________________________________________________________ summary ______________________________________________________________________________________
ERROR:   integration-test-nimbus: commands failed
```

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-1794)

--- a/app/tests/integration/nimbus/test_remote_settings.py
+++ b/app/tests/integration/nimbus/test_remote_settings.py
@@ -1,4 +1,5 @@
 import json
+import time
 
 import pytest
 import requests
@@ -7,25 +8,35 @@ from nimbus.pages.experimenter.summary import SummaryPage
 from nimbus.pages.remote_settings.dashboard import Dashboard
 from nimbus.pages.remote_settings.login import Login
 
+LOAD_DATA_RETRIES = 10
+LOAD_DATA_RETRY_DELAY = 1.0
+
 
 def load_data():
-    apps = []
-    data = requests.post(
-        "https://nginx/api/v5/graphql",
-        json={
-            "operationName": "getConfig",
-            "variables": {},
-            "query": "\nquery getConfig {\n  nimbusConfig "
-            "{\n    targetingConfigs {\n      "
-            "label\n      value\n      applicationValues\n    "
-            "}\n  }\n}\n",
-        },
-        verify=False,
-    ).json()
-    for item in data["data"]["nimbusConfig"]["targetingConfigs"]:
-        if "DESKTOP" in item["applicationValues"]:
-            apps.append(item["value"])
-    return apps
+    for retry in range(0, LOAD_DATA_RETRIES):
+        try:
+            apps = []
+            data = requests.post(
+                "https://nginx/api/v5/graphql",
+                json={
+                    "operationName": "getConfig",
+                    "variables": {},
+                    "query": "\nquery getConfig {\n  nimbusConfig "
+                    "{\n    targetingConfigs {\n      "
+                    "label\n      value\n      applicationValues\n    "
+                    "}\n  }\n}\n",
+                },
+                verify=False,
+            ).json()
+            for item in data["data"]["nimbusConfig"]["targetingConfigs"]:
+                if "DESKTOP" in item["applicationValues"]:
+                    apps.append(item["value"])
+            return apps
+        except json.JSONDecodeError:
+            if retry + 1 >= LOAD_DATA_RETRIES:
+                raise
+            else:
+                time.sleep(LOAD_DATA_RETRY_DELAY)
 
 
 @pytest.fixture(params=load_data())

